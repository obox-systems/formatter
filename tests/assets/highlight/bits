fn main() {
    let num1: u32 = 42;
    let num2: u32 = 23;

    let and_result = num1 & num2;
    println!("Bitwise AND: {}", and_result);

    let or_result = num1 | num2;
    println!("Bitwise OR: {}", or_result);

    let xor_result = num1 ^ num2;
    println!("Bitwise XOR: {}", xor_result);

    let shift_left_result = num1 << 2;
    println!("Shift Left: {}", shift_left_result);

    let shift_right_result = num2 >> 1;
    println!("Shift Right: {}", shift_right_result);

    let complement_result = !num1;
    println!("Bitwise NOT: {}", complement_result);

    let leading_zeros = num1.leading_zeros();
    println!("Leading Zeros: {}", leading_zeros);

    let trailing_zeros = num2.trailing_zeros();
    println!("Trailing Zeros: {}", trailing_zeros);
}
