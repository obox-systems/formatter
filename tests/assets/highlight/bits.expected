[42mfn[0m[47m [0m[42mmain[0m[100m([0m[100m)[0m[47m [0m[100m{[0m[101m
[0m[47m    [0m[42mlet[0m[47m [0m[42mnum1[0m[42m:[0m[47m [0m[42mu32[0m[47m [0m[42m=[0m[47m [0m[40m4[0m[40m2[0m[40m;[0m[101m
[0m[47m    [0m[42mlet[0m[47m [0m[42mnum2[0m[42m:[0m[47m [0m[42mu32[0m[47m [0m[42m=[0m[47m [0m[40m2[0m[40m3[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mand_result[0m[47m [0m[42m=[0m[47m [0m[42mnum1[0m[47m [0m[42m&[0m[47m [0m[42mnum2[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Bitwise AND: {}"[0m[40m,[0m[47m [0m[42mand_result[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mor_result[0m[47m [0m[42m=[0m[47m [0m[42mnum1[0m[47m [0m[40m|[0m[47m [0m[42mnum2[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Bitwise OR: {}"[0m[40m,[0m[47m [0m[42mor_result[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mxor_result[0m[47m [0m[42m=[0m[47m [0m[42mnum1[0m[47m [0m[40m^[0m[47m [0m[42mnum2[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Bitwise XOR: {}"[0m[40m,[0m[47m [0m[42mxor_result[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mshift_left_result[0m[47m [0m[42m=[0m[47m [0m[42mnum1[0m[47m [0m[42m<[0m[42m<[0m[47m [0m[40m2[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Shift Left: {}"[0m[40m,[0m[47m [0m[42mshift_left_result[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mshift_right_result[0m[47m [0m[42m=[0m[47m [0m[42mnum2[0m[47m [0m[40m>[0m[40m>[0m[47m [0m[40m1[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Shift Right: {}"[0m[40m,[0m[47m [0m[42mshift_right_result[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mcomplement_result[0m[47m [0m[42m=[0m[47m [0m[40m![0m[42mnum1[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Bitwise NOT: {}"[0m[40m,[0m[47m [0m[42mcomplement_result[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mleading_zeros[0m[47m [0m[42m=[0m[47m [0m[42mnum1[0m[40m.[0m[42mleading_zeros[0m[100m([0m[100m)[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Leading Zeros: {}"[0m[40m,[0m[47m [0m[42mleading_zeros[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mtrailing_zeros[0m[47m [0m[42m=[0m[47m [0m[42mnum2[0m[40m.[0m[42mtrailing_zeros[0m[100m([0m[100m)[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"Trailing Zeros: {}"[0m[40m,[0m[47m [0m[42mtrailing_zeros[0m[100m)[0m[40m;[0m[101m
[0m[100m}[0m[101m
[0m